wvd-templates/Create and provision WVD host pool/mainTemplate.json

```diff
@@ -292,81 +357,170 @@
+            }
+        },
+        {
+            "condition": "[variables('rdshManagedDisks')]",
+            "apiVersion": "2018-10-01",
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[concat(if(variables('rdshManagedDisks'), '', 'null'), variables('rdshPrefix'), copyindex())]",
+            "location": "[parameters('location')]",
+            "copy": {
+                "name": "rdsh-vm-loop",
+                "count": "[parameters('rdshNumberOfInstances')]"
+            },
+            "metadata": {
+                "comments": "If the following conditional dependencies are not deployed, ARM will auto remove them from this dependsOn list"
+            },
+            "dependsOn": [
+                "[concat('Microsoft.Compute/images/', variables('imageName'))]",
+                "[concat('Microsoft.Network/networkInterfaces/', variables('rdshPrefix'), copyindex(), '-nic')]"
+            ],
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('rdshVmSize')]"
+                },
+                "availabilitySet": "[if(not(parameters('enablePersistentDesktop')), main.getAvailabilitySet(variables('rdshPrefix')), json('null'))]",
+                "osProfile": {
+                    "computerName": "[concat(variables('rdshPrefix'), copyindex())]",
+                    "adminUsername": "[variables('existingDomainUsername')]",
+                    "adminPassword": "[parameters('existingDomainPassword')]"
+                },
+                "storageProfile": {
+                    "imageReference": "[if(equals(variables('rdshImageSourceLower'), 'customvhd'), main.getVMImageReferenceCustomVHD(variables('imageName')), variables('VMImageReference')[variables('rdshImageSourceLower')])]",
+                    "osDisk": {
+                        "createOption": "FromImage",
+                        "managedDisk": {
+                            "storageAccountType": "[parameters('rdshVMDiskType')]"
+                        }
+                    }
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(variables('rdshPrefix'), copyindex(), '-nic'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": false
+                    }
+                },
+                "licenseType": "Windows_Client"
+            }
+        },
+        {
+            "condition": "[not(variables('rdshManagedDisks'))]",
+            "apiVersion": "2018-10-01",
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[concat(if(variables('rdshManagedDisks'), 'null', ''), variables('rdshPrefix'), copyindex())]",
```

Comment:
This uses "null" as part of the name but later in this file all of the dependsOn references to a virtualMachine start with
`concat('Microsoft.Compute/virtualMachines/', variables('rdshPrefix')`
and I don't see how that will ever have "null" in the name; so it seems like none of the dependsOn would ever reference the name of this resource.

Reply:
The reason why I had to prepend it with `null` was that in order for the template to be valid, I have make the names of those 2 VM creation resources unique. Those resources with `null` will never be deployed because of the condition.

Reply:
Ah, I see, I didn't realize the position of the null was changed between this and the other virtual machine :)